### PR TITLE
general copy editing

### PR DESCRIPTION
**Pull request by drewpca**

### CODE REVIEW

File: expression-language.md [hunk 1]
```diff
@@ -125,25 +125,25 @@ literal ::= number-literal | boolean-literal | string-literal
 ```
 
 Because string literals can be either `''` or `""` delimited, in the former case, the `"` has to be escaped, while in
```
Comment (duglin) [inline expression-language.md:127]: isn't this backwards?

File: expression-language.md [hunk 2]
```diff
@@ -182,10 +182,10 @@ The type system contains 3 _primitive_ types:
   32-bit, twos-complement encoding.
 - _Boolean_: A boolean value of "true" or "false".
 
-The [types defined in the CloudEvents specification][ce-type-system] URI, URI Reference and Timestamp are represented as
+The [types defined in the CloudEvents specification][ce-type-system]-- URI, URI Reference, and Timestamp-- are represented as
```
Comment (duglin) [inline expression-language.md:185]: why "--" instead of just a comma? Looks a bit odd to me
image

Comment (duglin) [reply expression-language.md:185]: or parens

File: expression-language.md [hunk 3]
```diff
@@ -335,13 +335,13 @@ An evaluation might return an error together with the return value, which the ev
...
 
-When input parameters types of operator/function doesn't match the signatures, the CESQL engine MUST try to perform an
+When input parameters types of operator/function don't match the signatures, the CESQL engine MUST try to perform an
 implicit cast.
 
 Implicit casts must follow the same semantics of their equivalent explicit cast functions, as defined in
 [Casting and type checking](#351-casting-and-type-checking) sub-paragraph.
 
-We refer in this paragraph to **ambiguous** operator/function as an operator/function that is overloaded with another
+We define in this section an **ambiguous** operator/function as an operator/function that is overloaded with another
```
Comment (duglin) [inline expression-language.md:344]: This section defines an ....

File: expression-language.md [hunk 4]
```diff
@@ -40,27 +40,27 @@ This document is a working draft.
...
+The CloudEvents Expression Language assumes the input always includes, but is not limited to, a single valid and 
+type-checked CloudEvent instance. An expression MUST NOT mutate the value of the input CloudEvent instance, nor any of 
+the other input values. The evaluation of an expression observes the concept of [referential
 transparency][referential-transparency-wiki]. The output of a CESQL expression evaluation is always a boolean _true_ or
-_false_ and it might include an error.
+_false_, and it might include an error.
 
 The CloudEvents Expression Language doesn't support the handling of the data field of the CloudEvent instances, due to
-its polymorphic nature and complexity. We strongly encourage users that needs this functionality to use other more
+its polymorphic nature and complexity. We strongly encourage users that need this functionality use other more
```
Comment (duglin) [inline expression-language.md:63]: It is STRONGLY RECOMMENDED that users who need this...

Comment (duglin) [reply expression-language.md:63]: just trying to remove the use of "we"

Comment (drewpca) [reply expression-language.md]: As I understand the context, CESQL simply won't filter on the contents of `data`, so any implication that a user could pick CESQL (but shouldn't because we don't "recommend" it) seems misleading. Rewritten.

(Personally I don't get it. What's harder about letting me access fields in data? They could be missing sometimes, but that's already handled for the standard-but-optional fields.)

File: expression-language.md (hunk 3 above)
Comment (drewpca) [reply expression-language.md]: done

File: expression-language.md (hunk 2 above)
Comment (drewpca) [reply expression-language.md]: Parens lgtm.

File: expression-language.md (hunk 1 above)
Comment (drewpca) [reply expression-language.md]: Fixed

File: expression-language.md [hunk 5]
```diff
@@ -40,28 +40,27 @@ This document is a working draft.
...
+type-checked CloudEvent instance. An expression MUST NOT mutate the value of the input CloudEvent instance, nor any of 
+the other input values. The evaluation of an expression observes the concept of [referential
 transparency][referential-transparency-wiki]. The output of a CESQL expression evaluation is always a boolean _true_ or
-_false_ and it might include an error.
+_false_, and it might include an error.
 
 The CloudEvents Expression Language doesn't support the handling of the data field of the CloudEvent instances, due to
-its polymorphic nature and complexity. We strongly encourage users that needs this functionality to use other more
-appropriate tools.
+its polymorphic nature and complexity. Users that need this functionality should use other more appropriate tools.
```
Comment (duglin) [inline expression-language.md:63]: s/should/SHOULD/

File: expression-language.md [hunk 6]
```diff
@@ -40,28 +40,27 @@ This document is a working draft.
...
+type-checked CloudEvent instance. An expression MUST NOT mutate the value of the input CloudEvent instance, nor any of 
+the other input values. The evaluation of an expression observes the concept of [referential
 transparency][referential-transparency-wiki]. The output of a CESQL expression evaluation is always a boolean _true_ or
-_false_ and it might include an error.
+_false_, and it might include an error.
 
 The CloudEvents Expression Language doesn't support the handling of the data field of the CloudEvent instances, due to
-its polymorphic nature and complexity. We strongly encourage users that needs this functionality to use other more
-appropriate tools.
+its polymorphic nature and complexity. Users that need this functionality SHOULD use other more appropriate tools.
```
Comment (slinkydeveloper) [inline expression-language.md:63]: Do we need this `SHOULD` normative keyword here @duglin? This is an informal description/introduction, it doesn't explain the spec behaviour...

File: expression-language.md [hunk 7]
```diff
@@ -111,7 +110,7 @@ lowercase-char ::= [a-z]
 value-identifier ::= ( lowercase-char | digit ) ( lowercase-char | digit )*
 ```
 
-CESQL defines 3 different literal kinds: integer numbers, `true` or `false` booleans and `''` or `""` delimited strings.
+CESQL defines 3 different literal kinds: integer numbers, `true` or `false` booleans, and `''` or `""` delimited strings.
```
Comment (slinkydeveloper) [inline expression-language.md:113]: I'm quite sure you're better than me with english :), but is the comma here required?

File: expression-language.md [hunk 8]
```diff
@@ -182,10 +181,10 @@ The type system contains 3 _primitive_ types:
   32-bit, twos-complement encoding.
 - _Boolean_: A boolean value of "true" or "false".
 
-The [types defined in the CloudEvents specification][ce-type-system] URI, URI Reference and Timestamp are represented as
+The [types defined in the CloudEvents specification][ce-type-system] (URI, URI Reference, and Timestamp) are represented as
```
Comment (slinkydeveloper) [inline expression-language.md:184]: I don't think we need parenthesis here, same for the last comma you added

**Review by slinkydeveloper — COMMENTED** [review]
Thanks for this contribution! Did you run prettier with wrapping at 120 after all the typos?

**Comment by duglin** [comment]
please sign your commits so the DCO checker is happy

File: expression-language.md (hunk 7 above)
Comment (drewpca) [reply expression-language.md:113]: https://en.wikipedia.org/wiki/Serial_comma 

Style guides vary, but note that we already have it on 54, 131, 184, 268. It's usually the less-ambiguous choice.

File: expression-language.md (hunk 8 above)
Comment (drewpca) [reply expression-language.md]: It's a garden-path sentence to have "... defined in the CloudEvents specification URI, ...", since the specification has a URI.

I see though, that this isn't just a reminder of what the 3 types are. It's a rule applying to only those types. I reordered the sentence to get the emphasis right.

**Comment by drewpca** [comment]
I can't figure out how to sync this.

```
~/Downloads/spec% git remote -v
origin	git@github.com:drewpca/spec.git (fetch)
origin	git@github.com:drewpca/spec.git (push)

~/Downloads/spec% git push origin patch-1
To github.com:drewpca/spec.git
 ! [rejected]        patch-1 -> patch-1 (non-fast-forward)
error: failed to push some refs to 'github.com:drewpca/spec.git'
hint: Updates were rejected because a pushed branch tip is behind its remote
hint: counterpart. Check out this branch and integrate the remote changes
hint: (e.g. 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

~/Downloads/spec% git pull origin patch-1 

From github.com:drewpca/spec
 * branch            patch-1    -> FETCH_HEAD
hint: Pulling without specifying how to reconcile divergent branches is
hint: discouraged. You can squelch this message by running one of the following
hint: commands sometime before your next pull:
hint: 
hint:   git config pull.rebase false  # merge (the default strategy)
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint: 
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.
Merge made by the 'recursive' strategy.

~/Downloads/spec% git push origin patch-1
To github.com:drewpca/spec.git
 ! [rejected]        patch-1 -> patch-1 (non-fast-forward)
error: failed to push some refs to 'github.com:drewpca/spec.git'
hint: Updates were rejected because a pushed branch tip is behind its remote
hint: counterpart. Check out this branch and integrate the remote changes
hint: (e.g. 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

**Review by slinkydeveloper — APPROVED** [review]

**Comment by drewpca** [comment]
I got tired of fighting git/github and lost interest. Please do take whatever is still relevant. I don't need my name on the commits or anything.

File: cesql/spec.md [hunk 9]
```diff
@@ -314,8 +313,6 @@ For example, the following definitions are valid:
 * `ABC(x, y): String x String -> Integer`: Binary function (arity 2).
 * `ABC(x, y, z, ...): String x String x String x String^n -> Integer`: n-ary function (variable arity), but the initial fixed arguments are at least 3.
 
-But the followings are invalid, so the engine MUST reject them:
```
Comment (duglin) [inline cesql/spec.md:317]: @drewpca why did you remove this line?

**Comment by duglin** [comment]
@drewpca I squads and rebased it - look it over and see if I missed anything. Also I asked a question about one line: https://github.com/cloudevents/spec/pull/788#discussion_r779707733

**Comment by duglin** [comment]
All, please review the latest as I'll be adding it to the agenda for the next call

File: cesql/spec.md (hunk 9 above)
Comment (drewpca) [reply cesql/spec.md:317]: Looks like a merge error. I see no reason to remove that line

Comment (duglin) [reply cesql/spec.md:317]: ok thanks - I'll add it back in

**Comment by duglin** [comment]
Approved on the 1/13 call
